openedx_external_enrollments/external_enrollments/viper_external_enrollment.py

```diff
@@ -0,0 +1,133 @@
+"""ViperExternalEnrollment class file."""
+import logging
+
+import requests
+from django.conf import settings
+from django.core.exceptions import ObjectDoesNotExist
+from rest_framework import status
+
+from openedx_external_enrollments.external_enrollments.base_external_enrollment import BaseExternalEnrollment
+from openedx_external_enrollments.models import ExternalEnrollment
+
+LOG = logging.getLogger(__name__)
+
+
+class ViperExternalEnrollment(BaseExternalEnrollment):
+    """
+    ViperExternalEnrollment class.
+    """
+    CREATE_ENROLLMENT_ACTION = 'createEnrolment'
+
+    def __str__(self):
+        return 'viper'
+
+    def _execute_post(self, url, data=None, headers=None, json_data=None):
+        """
+        Execute post request.
+        """
+        course_shell_id = json_data.get('variables').pop('shellCourseId')
+        response = requests.post(
+            url=url,
+            json=json_data,
+            headers=headers,
+        )
+        enrollment_data = json_data.get('variables')
```

Comment:
This line should be inside the following if, since it's the only scope for `enrollment_data` .